components/script/dom/imagedata.rs

```diff
@@ -144,23 +147,23 @@ impl ImageData {
         width: u32,
         opt_height: Option<u32>,
     ) -> Fallible<DomRoot<Self>> {
-        Self::new_with_jsobject(global, proto, width, opt_height, jsobject)
+        // Self::new_with_data(global, proto, width, opt_height, jsobject)
+        Self::new_without_data(global, proto, width, opt_height.unwrap())
```

Comment:
Here you should keep `new_with_jsobject`, and create a new `new_initialized` method on `HeapTypedArray` that takes an optional jsobject, in this case some.